ReactNativeClient/lib/joplin-database.js

```diff
@@ -735,6 +735,73 @@ class JoplinDatabase extends Database {
 				);
 			}
 
+			if (targetVersion == 31) {
+				queries.push('DROP TRIGGER IF EXISTS notes_fts_before_update');
+				queries.push('DROP TRIGGER IF EXISTS notes_fts_before_delete');
+				queries.push('DROP TRIGGER IF EXISTS notes_after_update');
+				queries.push('DROP TRIGGER IF EXISTS notes_after_insert');
```

Comment:
Why would they exist since you are creating them in this migration?

We should assume that the database is in a consistent state when running migrations. Trying to handle every possible cases make things more complex and can cause subtle bugs. I'd rather the migration fails if one of the table already exists than deleting data without really knowing what's going on.

Reply:
These triggers are created in `targetVersion == 15`. But I'll remove the `IF EXISTS`.